modules/SofaMiscMapping/CenterOfMassMapping.inl

```diff
@@ -46,15 +46,22 @@ template <class TIn, class TOut>
 void CenterOfMassMapping<TIn, TOut>::init()
 {
     //get the pointer on the input dofs mass
-    masses = this->fromModel->getContext()->getMass();
-    if(!masses)
+    if (!l_mass)
+    {
+        l_mass.set(this->fromModel->getContext()->getMass());
```

Comment:
I would add a warning here telling: mass link need to be set to ensure right behavior. First mass component found in the context will be used.

Reply:
:+1:  see https://github.com/sofa-framework/sofa/pull/1183/files#diff-528b3d11ee0d011535934ce92aaac5ccR153-R159